modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4TcpChannel.java

```diff
@@ -61,6 +59,24 @@
                 }
             }
         });
+
+        if (connectFuture != null) {
```

Comment:
can we just not make it nullable? it would be nice to have it just be there all the time?